core/style.js

```diff
@@ -1395,7 +1395,7 @@ CKEDITOR.STYLE_OBJECT = 3;
 		if ( CKEDITOR.env.ie )
 			preBlock.$.outerHTML = '<pre>' + mergedHtml + '</pre>';
 		else
-			preBlock.setHtml( mergedHtml );
+			preBlock.setHtml( CKEDITOR.tools.htmlSafeByReview(mergedHtml, 'preBlock is created internally and is only impacted by the editors config options') );
```

Comment:
TODO: look into preblocks and injection

We can potentially sanitize here.

Reply:
Looked deeper into mergedHtml. It is created from preBlocks which go through a chain of internal functions before reaching removeFromRange( range ) where range is some set of document dom elements. The CKEditor DOM api is only accessible to the developer or used internally so there shouldn't be a threat of injection.

Reply:
Legacy converted